nping/utils.cc

```diff
@@ -316,26 +316,17 @@ int bitcmp(u8 *a, u8*b, int len){
 
 /** Removes every instance of the character stored in parameter "c" in the
  * supplied string.
- * @warning the supplied buffer is modified by this function. Whenever a
- * colon is found, the rest of the string is moved one position to the left
- * so the colon gets overwritten. */
+ * @warning the supplied buffer is modified by this function. */
 int removechar(char *string, char c){
   size_t len=0, i=0, j=0;
   if(string==NULL)
     return OP_FAILURE;
   len=strlen(string);
```

Comment:
Why get the length with `strlen` if you will be checking every index anyway? Just change the `while` condition to check for `'\0'`, which signals the end of the string.